runtime/protocol/http-client-engines/http-client-engine-crt/jvm/src/aws/smithy/kotlin/runtime/http/engine/crt/SdkStreamResponseHandler.kt

```diff
@@ -210,4 +252,19 @@ internal class SdkStreamResponseHandler(
             conn.close()
         }
     }
+
+    override fun onMetrics(stream: HttpStream, metrics: HttpStreamMetrics) {
+        val sendEnd = positiveInstantOrNull(metrics.sendEndTimestampNs)
+        val receiveStart = positiveInstantOrNull(metrics.receiveStartTimestampNs)
+
+        if (sendEnd != null && receiveStart != null) {
+            val ttfb = receiveStart - sendEnd
+            if (ttfb.isPositive()) {
```

Comment:
Not directly but event streams may begin receiving traffic before sending has concluded since the communication is bidirectional and duplexed. It seemed prudent in that situation to not report TTFB.